### PR TITLE
Bigtable: add await replication

### PR DESCRIPTION
**Pull request by igorbernstein2**
This replaces the raw calls to generate/check consistancy token with a
polling wrapper that wait until all clusters are consistent.

### CODE REVIEW

**Comment by igorbernstein2** [comment]
@kevinsi4508 please take a look when you have a moment

**Comment by igorbernstein2** [comment]
I split off the stub separation here:
https://github.com/GoogleCloudPlatform/google-cloud-java/pull/3691

I can't think of a reason to expose the raw rpcs for generating tokens & checking them. Current implementations in go & java find the raw methods as well

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.MoreExecutors;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CancellationException;
+
+/**
+ * Callable that waits until replication has caught up to the point it was called.
+ *
+ * <p>This callable wraps GenerateConsistencyToken and CheckConsistency RPCs. It will generate
+ * a token then poll until it isConsistent is true.
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java]: done

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+class AwaitReplicationCallable extends UnaryCallable<TableName, Void> {
+  private final UnaryCallable<GenerateConsistencyTokenRequest, GenerateConsistencyTokenResponse> generateCallable;
+  private final UnaryCallable<CheckConsistencyRequest, CheckConsistencyResponse> checkCallable;
+  private final RetryingExecutor<CheckConsistencyResponse> executor;
+
+  static AwaitReplicationCallable create(
+      UnaryCallable<GenerateConsistencyTokenRequest, GenerateConsistencyTokenResponse> generateCallable,
+      UnaryCallable<CheckConsistencyRequest, CheckConsistencyResponse> checkCallable,
+      ClientContext clientContext,
+      RetrySettings pollingSettings) {
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java]: Gax uses retry settings for both polling and retries. I would like to call out that these settings are a bit different than general retries

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    @Override
+    public TimedAttemptSettings createNextAttempt(Throwable prevThrowable,
+        CheckConsistencyResponse prevResponse, TimedAttemptSettings prevSettings) {
+      return null;
+    }
+
+    @Override
+    public boolean shouldRetry(Throwable prevThrowable, CheckConsistencyResponse prevResponse)
+        throws CancellationException {
+
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallable.java]: done

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/EnhancedBigtableTableAdminStub.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    RetrySettings pollingSettings = RetrySettings.newBuilder()
+        // use overall timeout from checkConsistencyCallable
+        // NOTE: The overall timeout might exceed this value due to underlying retries
+        .setTotalTimeout(settings.checkConsistencySettings().getRetrySettings().getTotalTimeout())
+        // Use constant polling with jitter
+        .setInitialRetryDelay(Duration.ofSeconds(10))
+        .setRetryDelayMultiplier(1.0)
+        .setMaxRetryDelay(Duration.ofSeconds(10))
+        .setJittered(true)
+        // rpc timeouts are ignored
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/stub/EnhancedBigtableTableAdminStub.java]: updated the comment

**Comment by igorbernstein2** [comment]
Thanks for reviewing! @pongad can you take at this after #3691?

**Comment by pongad** [comment]
@igorbernstein2 merging #3691 seems to have caused some conflicts, could you resolve?

**Comment by igorbernstein2** [comment]
Should be good to go now

**Pull request merged**